app/src/main/java/com/hedvig/app/feature/offer/OfferViewModel.kt

```diff
@@ -374,32 +384,30 @@ class OfferViewModelImpl(
         offerRepository.writeDiscountToCache(quoteIds, data)
 
     override suspend fun triggerOpenChat() {
-        val event = when (chatRepository.triggerFreeTextChat()) {
-            is Either.Left -> Event.Error
-            is Either.Right -> Event.OpenChat
-        }
-        _events.trySend(event)
+        chatRepository.triggerFreeTextChat()
+            .fold(
+                ifLeft = { Event.Error },
+                ifRight = { Event.OpenChat }
+            )
+            .let(_events::trySend)
     }
 
     override fun onOpenQuoteDetails(id: String) {
         viewModelScope.launch {
-            when (val result = getQuoteUseCase.invoke(quoteIds, id)) {
-                GetQuoteUseCase.Result.Error -> {
-                    offerAndLoginStatus.value = OfferAndLoginStatus.Error
-                }
-                is GetQuoteUseCase.Result.Success -> {
+            getQuoteUseCase.invoke(quoteIds, id)
+                .mapLeft { offerAndLoginStatus.value = OfferAndLoginStatus.Error }
+                .map {
                     _events.trySend(
                         Event.OpenQuoteDetails(
                             QuoteDetailItems(
-                                result.quote.displayName,
-                                result.quote.perils.map { PerilItem.Peril(it) },
-                                result.quote.insurableLimits,
-                                result.quote.insuranceTerms
+                                it.displayName,
+                                it.perils.map { PerilItem.Peril(it) },
+                                it.insurableLimits,
+                                it.insuranceTerms
                             )
                         )
                     )
                 }
```

Comment:
Hmm I think this one is a bit different from how we use Either in the other cases.
Specifically previously if we wanted to simply do something on left and on right we used fold to do this, but here we have a mapLeft and a map.
This is something I'm still trying to figure out in general how to consistently do things in a certain way, but I have some comments.
Here if we just say we won't do fold() on, the mapLeft feels like it should be a `tapLeft` instead, since with this after the line of mapLeft we changed the type of Either from `Either<A, B>` to `Either<Unit, B>` since we mapped the left to the result of `offerAndLoginStatus.value = OfferAndLoginStatus.Error` which is a variable assignment which I assume would be `Unit`. What we probably were going for was the idea of just "do something if it's left" which `tap|tapLeft` exists for afaik.
Same exact idea with `map`, where we were looking for just doing `tap` since we again don't then need to furhter change the type from `Either<Unit, B>` to `Either<Unit, ChannelResult<Unit>>` which we are here since map will change the type with the last result in the lambda which here is the result of trySend which is the type I said above.

Because this is something that I've been thinking in general I asked [this](https://kotlinlang.slack.com/archives/C5UPMM0A0/p1646391903231499) to potentially get some input from the Arrow community as well. Maybe we can se how people in general like to do this.